core/src/providers/google_vertex_ai.rs

```diff
@@ -20,24 +20,74 @@ use crate::{
 
 use super::{
     embedder::Embedder,
-    llm::{ChatFunction, ChatMessage, LLMChatGeneration, LLMGeneration, LLM},
+    llm::{ChatFunction, ChatFunctionCall, ChatMessage, LLMChatGeneration, LLMGeneration, LLM},
     provider::{Provider, ProviderID},
     tiktoken::tiktoken::{
         cl100k_base_singleton, decode_async, encode_async, tokenize_async, CoreBPE,
     },
 };
 
+// Disabled for now as it requires using a "tools" API which we don't support yet.
+pub const USE_FUNCTION_CALLING: bool = false;
```

Comment:
Our implementation of it works. But using it means we cannot use it with assistants in their current form as the way we use function calling is not compatible with the "tools" API of Gemini